DataDog/datadog-operator
# Support custom e2e kustomization resources

### What does this PR do?

* Support adding custom resources in `config/e2e/kustomization.yaml` in E2E kind provisioner. 
* Updates `config/e2e/kustomization.yaml` with latest operator image name and tag if `IMG` environment variable is not set
* Enables E2E tests for merge requests to main

### Motivation

Provide more operator customization for E2E tests and add defaults for E2E operator image.

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

* Run `IMG=<custom_operator_image> aws-vault exec sso-agent-sandbox-account-admin -- make e2e-tests` 
  * Verify that `config/e2e/kustomization.yaml` is updated with the correct controller image name and tag
  * Verify that E2E tests pass
* Run `aws-vault exec sso-agent-sandbox-account-admin -- make e2e-tests` 
  * Verify that `config/e2e/kustomization.yaml` is updated with the default `gcr.io/datadoghq/operator` image name and `latest` tag
  * Verify that E2E tests pass


### Checklist
